pyatrea/__init__.py

```diff
@@ -1,226 +1,474 @@
+                                self.status[child.attrib["I"]] = child.attrib["V"]
+        return self.status
 
     def getTranslation(self, id):
         translations = self.getTranslations()
-        
-        if(id in translations['params']):
-            param = 'params'
-        elif(id in translations['words']):
-            param = 'words'
+
+        if id in translations["params"]:
+            param = "params"
+        elif id in translations["words"]:
+            param = "words"
         else:
             return id
 
-        if('d' in translations[param][id]):
-            toBeTranslated = translations[param][id]['d']
-            if(toBeTranslated == "not%20to%20be%20translated"):
-                toBeTranslated = translations[param][id]['t']
+        if "d" in translations[param][id]:
+            toBeTranslated = translations[param][id]["d"]
+            if toBeTranslated == "not%20to%20be%20translated":
+                toBeTranslated = translations[param][id]["t"]
         else:
-            toBeTranslated = translations[param][id]['t']
+            toBeTranslated = translations[param][id]["t"]
 
-        if(hasattr(urllib, 'parse')):
+        if hasattr(urllib, "parse"):
             return urllib.parse.unquote(toBeTranslated)
         else:
-            return urllib.unquote(toBeTranslated) #pylint: disable=E1101
-    
+            return urllib.unquote(toBeTranslated)  # pylint: disable=E1101
+
     def loadSupportedModes(self):
         status = self.getStatus()
-        if(status == False):
-            return False
-        try:
-            binary_writable_modes = '{0:08b}'.format(int(status['I12004']))
-            H11700 = int(status['H11700'])
-        except AttributeError:
+        if status == False:
             return False
-            
-        for i in range(8):
-            if ((i == 3 or i == 4) and (int(H11700) == 0)):
-                self.writable_modes[i] = False
-            else:
-                if(int(binary_writable_modes[7-i]) == 0):
+
+        for mode in list(AtreaMode):
+            self.writable_modes[mode] = False
+
+        if "I12004" in status and "H11700" in status:
+            try:
+                binary_writable_modes = "{0:08b}".format(int(status["I12004"]))
+                H11700 = int(status["H11700"])
+            except AttributeError:
+                return False
+
+            for i in range(8):
+                if (i == 3 or i == 4) and (int(H11700) == 0):
                     self.writable_modes[i] = False
                 else:
-                    self.writable_modes[i] = True
+                    self.writable_modes[i] = int(binary_writable_modes[7 - i]) != 0
+        else:
+            response = requests.get(
+                "http://"
+                + self.ip
+                + "/lang/userCtrl.xml?auth="
+                + self.code
+                + random.choice(string.ascii_letters)
+                + random.choice(string.ascii_letters)
+            )
```

Comment:
This should be:

```
response = requests.get(
                "http://"
                + self.ip
                + "/lang/userCtrl.xml?auth="
                + self.code
                + "&"
                + random.choice(string.ascii_letters)
                + random.choice(string.ascii_letters)
            )
```

Reply:
Fixed. Thanks for catching that, my unit does not support auth, so I didn't catch it during the testing.